apps/condo/domains/notification/adapters/smsAdapter.js

```diff
@@ -58,6 +66,122 @@ class SMSAdapter {
     }
 }
 
+class InstasentSms {
+
+    isConfigured = false
+
+    constructor () {
+        let config = conf['INSTASENT_API_CONFIG'] ?  JSON.parse(conf['INSTASENT_API_CONFIG']) : {}
+        this.isConfigured = validateConfig(config, [
+            'apiKey',
+        ])
+        this.apiKey = config.apiKey
+    }
+
+    isPhoneSupported (phoneNumber) {
+        return AllCountriesTest.test(phoneNumber)
+    }
+
+    async checkIsAvailable () {
+        const result = await fetch(
+            'https://api.instasent.com/organization/account',
+            {
+                method: 'GET',
+                headers: {
+                    'Authorization': `Bearer ${this.apiKey}`,
+                    'Accept': 'application/json',
+                },
+            }
+        )
+        return result.ok
+    }
+
+    async send ({ phone, message }, extendedParams = {}) {
+        const body = {
+            clientId: crypto.randomBytes(16).toString('hex'),
```

Comment:
Strange behaviour of client id generation for every single request

Reply:
Maybe something `sender.fingerprint` can be used here?